src/common/task.c

```diff
@@ -259,6 +265,10 @@ int64_t task_args_add_val(task_spec *spec, uint8_t *data, int64_t length) {
   return spec->arg_index++;
 }
 
+void task_add_required_resource(task_spec *spec, int64_t resource_index, double value) {
+  spec->required_resources[resource_index] = value;
```

Comment:
Should probably do some error checking on `value` (e.g., `PyFloat_AsDouble` can fail and return -1). Also, are there required resources that wouldn't have an integer value?